src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -543,7 +544,8 @@ type DiagnosticItem =
             | ErrorCode.ExpectingIterableExpr                     -> "The type {0} does not support iteration. Expecting an expression of array type or of type Range."
             | ErrorCode.ExpectingCallableExpr                     -> "The type of the expression must be a function or operation type. The given expression is of type {0}." 
             | ErrorCode.UnknownIdentifier                         -> "No identifier with the name \"{0}\" exists."
-                                                            
+            | ErrorCode.UnsupportedResultComparison               -> "Measurement results cannot be compared because the execution target supports only {0} runtime capabilities."
```

Comment:
My philosophy on error messages is that they should be as specific as possible and contain actionable suggestions if possible. Here is my ideal but pretty long error message:

> Measurement results cannot be compared because the execution target is {0}, which supports only {1} runtime capabilities. Choose an execution target with QPRGen1 capabilities to compare results within an if-condition, or Unknown capabilities to compare results anywhere.

I want to include specific details about the project settings in case the user does not remember what execution target they have set (or is not aware that they set one - e.g. if the execution target was set by a project template). At minimum, this would be either the target ID or its runtime capabilities (or both).

The name of the execution target isn't available where the error message is generated right now, so I would have to figure out a way to get it there. Do you think it's worth it?

Reply:
I agree that actionable error messages are best, but like Bettina said the "QPRGen1" term might just be more confusing until we have better documentation for it. That makes it a bit of an annoying thing, because once we do have docs who is going to remember to come back and update these messages? Maybe a middle ground for now would be "Measurement results cannot be compared because the execution target {0} does have that capability. Choose an execution target with additional capabilities or avoid usage of measurement comparison." Then pair it with a nice todo comment and/or issue in the backlog to improve the error message once we know how we are going to document what capabilities exist and which providers have which capabilities.

Reply:
I am happy both with exposing the execution target, or also remain the runtime capabilities to something like "NoControlFlow" and "LimitedControlFlow". Just the QPRGen* is not really something that should be exposed.

Reply:
I added a TODO for including the runtime capabilities in the error message. The message is now:

> The execution target {0} does not support comparing measurement results. Choose an execution target with additional capabilities or use a controlled operation instead.